scheduler/test/cook/test/mesos/util.clj

```diff
@@ -160,16 +160,65 @@
            job-ent (d/entity @(d/sync conn) job)]
         (is (= (util/job-ent->attempts-consumed db job-ent) 2))
        ))
-    
+
     ))
 
 (deftest test-namespace-datomic
   (testing "Example tests"
     (is (= (util/namespace-datomic :straggler-handling :type)
            :straggler-handling/type))
     (is (= (util/namespace-datomic :straggler-handling :type :quantile-deviation)
-           :straggler-handling.type/quantile-deviation ))
-    )
-  )
+           :straggler-handling.type/quantile-deviation))))
+
+(deftest test-clear-uncommitted-jobs
+  (testing "no uncommitted"
+    (let [uri "datomic:mem://test-clear-uncommitted"
+          conn (restore-fresh-database! uri)
+          _ (dotimes [_ 100]
+              (create-dummy-job conn))]
+      (is (= (count (util/clear-uncommitted-jobs conn (t/yesterday) true)) 0))))
+  (testing "uncommitted but not before"
+    (let [uri "datomic:mem://test-clear-uncommitted"
+          conn (restore-fresh-database! uri)
+          _ (dotimes [_ 100]
+              (create-dummy-job conn))
+          _ (dotimes [_ 100]
+              (create-dummy-job conn
+                                :committed? false
+                                :submit-time (tc/to-date (-> 6 t/hours t/ago))))]
+      (is (= (count (util/clear-uncommitted-jobs conn (t/yesterday) true)) 0))))
+  (testing "uncommitted dry-run"
+    (let [uri "datomic:mem://test-clear-uncommitted"
+          conn (restore-fresh-database! uri)
+          _ (dotimes [_ 100]
+              (create-dummy-job conn))
+          _ (dotimes [_ 100]
+              (create-dummy-job conn
+                                :committed? false
+                                :submit-time (tc/to-date (-> 6 t/hours t/ago))))
+          uncommitted-count 25
+          _ (dotimes [_ uncommitted-count]
+              (create-dummy-job conn
+                                :committed? false
+                                :submit-time (tc/to-date (-> 48 t/hours t/ago))))]
+      (is (= (count (util/clear-uncommitted-jobs conn (t/yesterday) true))
+             uncommitted-count))))
+  (testing "uncommitted with retract"
```

Comment:
I'm indifferent. I split them so that we could test that dry run did the thing we expect.